javascript/javascript1/week2/candy_helper.js

```diff
@@ -0,0 +1,38 @@
+const boughtCandies = []
+let totalAmount = 0;
+function addCandy (candyType, weight) {
+    if (candyType === "sweet") {
+        boughtCandies.push(weight*0.5);
+    } else if (candyType === "chocolate") {
+        boughtCandies.push(weight*0.7);
+    } else if (candyType === "toffee") {
+        boughtCandies.push(weight*1.1);
+    } else if (candyType === "chewing-gum") {
+        boughtCandies.push(weight*0.03);
+    }
+}
+
+function totalPrice () {
+    for (let i=0; i<boughtCandies.length; i++) {
+        totalAmount += boughtCandies[i];
+    }
+    console.log("Final Price: $" + totalAmount);
```

Comment:
This is exactly how it had to be done. There are definitely other ways, but in this case this is the most common approach. Well done.

Reply:
Hi, Yasen. First of all, thank you for taking the time to make all those comments. I truly appreciate your help. I've read your suggestions through and I've applied all of them. 

One of the things I'm not really sure about yet is how to make sure Prettier works, as I don't see much difference. I've taken a look at the link you sent and set Prettier to format on save. 

Regarding the event_application exercise, I've changed the array starting from Sunday, not Monday, as I noticed new Date().getDay() returns 0 for Sunday. I think if I had chosen Monday to be the first element of the array, the code wouldn't have worked properly. Please, correct me if I'm wrong. 

When it comes to the student_manager exercise, I've tried to add more students WITHIN the class07students array, and even though I won't be able to add more students by calling the function and passing new arguments, the array does accept more than 6 students outside of the function. This is absolutely expectable, but I'm not sure if it follows the task's instructions. 
EDIT: To solve this I created a new function using the .splice( ) method to delete all the elements from the array starting from the 7th.

A question I've come across while doing these exercises is the when to choose console.log( ) or return in a function. Is it correct to say that if I don't want the output of that function to be shown, I should use return?

I'll come to you today to check all of this, so don't worry about answering :) 
Thank you again for all the positive comments on the code! 💪💪💪

Reply:
Sorry, I saw your message late. We could have went through it in class, but here is my reply.

A quick way to test if Prettier works is to remove all indentation from a file and then save it. If the file suddenly adds back the correct indentation, then Prettier works. Or if you add extra empty lines after the last line of code - Prettier would remove all, but the last one empty.

That's a very good reason and argument for the event_application to keep Sunday as the first day of the week. All good.

We use a lot of console.log now, but the idea is only for us as developers to test that the code works and we cover all edge cases. When you work on a real project, you remove all console logs at the end, as you don't want to give away what or how your code works exactly. Functions are often called through a variable, which then is console-logged. In this case, it doesn't make a big difference if you are console logining to the function or outside of it. Rasmus, however, suggested that we avoid as much as possible logging in functions, as you never know when they will be called and how many times, as that might clutter the terminal. Sorry, that I can't give a more definite answer.

I didn't see a problem with the student task. I tried to add a couple more students with the function and it worked fine. The task instructions were to add students in a way that you can test the various edge cases. I see you test for the limit of students "Cannot add more students to class 07" and "There are 7 in class 07. See the full list of students below: ...". May be add few more instances like those

```
addStudentToClass("Micah");
addStudentToClass("Steve");
addStudentToClass("Jason");
addStudentToClass("    ");
addStudentToClass("");
addStudentToClass("Bob");
addStudentToClass("Jason");
```
so you can log all other possible cases.

Reply:
Thanks for your reply Yasen! My biggest question was regarding return and console.log in a function, and after last Sunday it's much clearer! 

See you tomorrow! 🙌🏻